### ISSUE TITLE
useWaitForTransactionReceipt - onReplaced not triggering

### ISSUE THREAD

**Issue by tjelvar-chromaway** [wevm/wagmi#4226]
### Check existing issues

- [X] I checked there isn't [already an issue](https://github.com/wevm/wagmi/issues) for the bug I encountered.

### Describe the bug

I want to be able to react to replaced transactions. For example when a user speeds them up in metamask I would expect `onReplaced` to trigger with `replacement.reason` set to `repriced ` but currently the callback does not trigger at all. 

```typescript
const receipt = useWaitForTransactionReceipt({
    hash,
    onReplaced: (replacement) => {
      console.log(replacement);
    },
  });
```
I would expect the above code to work.

### What Wagmi package(s) are you using?

wagmi

### Wagmi Package(s) Version(s)

2.12.7

### Viem Version

2.9.13

### TypeScript Version

5.3.2

**Comment by dzek69** [comment]
I can confirm this does not work for me either.

**Comment by tjelvar-chromaway** [comment]
I found that wether or not `onReplaced` is triggered depends on how `transports` is setup. When using the default `http()`  it did trigger however, when i used `https://bsc-testnet-rpc.publicnode.com` it did not. @tmm is there some documentation on which requirements the RPC node needs to fulfil in order to support `onReplaced` being triggered? 

Here is the stackblitz: https://stackblitz.com/edit/vitejs-vite-jzuwmt?file=src%2FApp.tsx

**Comment by tjelvar-chromaway** [comment]
@tmm i've tried with paid ones now as well with no luck :/ 

This configuration that uses the bundled RPC node  triggers `onReplaced` _most of the times_

```typescript
import * as wagmiChains from "wagmi/chains";
import { createConfig, http } from "wagmi";

const bscTestnet = wagmiChains.bscTestnet;
export const onReplacedCompatibleConfig = createConfig({
  chains: [bscTestnet],
  transports: {
    [bscTestnet.id]: http(), // using bundled RPC
  },
});

const receipt = useWaitForTransactionReceipt({
  config: onReplacedCompatibleConfig,
  hash,
  onReplaced: (replacement) => {
    console.log(replacement);
  },
});
```

**Comment by dzek69** [comment]
> This configuration that uses the bundled RPC node triggers onReplaced

Sometimes.

@tmm: What do I look for if I want to make sure that RPC correctly supports `onReplaced`? Cause I used a private/paid one from quicknode.com and it does not trigger onReplaced as well (at least on BSC Testnet, I noticed many free ones works with ~90% of success on BSC Mainnet, but it's super rare/impossible for any RPC to trigger onReplaced on BSC Testnet).
